[BTS-49] 랜딩페이지

mozzi34 opened the pull request:
## ✅ 체크리스트
- [x] 코드에 린트 에러가 없습니다.
- [x] 코드가 올바르게 포맷팅되었습니다.
- [x] 빌드가 올바르게 실행되었습니다.

## 📌 변경 사항
  
- 랜딩페이지 구현
- route 객체 생성

지금은 색상/폰트 크기 관련 공통 파일이 없어서 일단 직접 작성했습니다 관련 파일이 정리되면 일괄 수정하도록 하겠습니다 또한 버튼과 같은 공통 컴포넌트도 직접 작성했지만 후에 수정 예정입니다!

## 🧐 관련 이슈
BTS-49: 홈페이지 구현

## 📷 스크린샷 or 코드 (선택사항)

## 📚 참고 자료

nononcrust committed:
chore: 프로젝트 설정 수정 (#3)

* tsconfig noUncheckedIndexedAccess 옵션 추가

* endOfLine auto -> lf

* check-types:watch 스크립트 추가

* styles 파일 위치 변경, tailwindcss 기본 스타일 제거

* nextjs 기본 파일 제거

Commit:
chore: 프로젝트 ci 설정 (#4)

* chore: main 브랜치를 개인 레포로 동기화 + 린트/타입 검사

* fix: 오타 수정

* chore(ci): 워크플로우 파일명 'ci-lint-build.yml'로 변경 및 build 스크립트 추가

nononcrust committed:
chore: tailwind-merge 패키지 설치, api 인스턴스, cn 함수 추가

mozzi34 committed:
chore: 랜딩페이지용 SVG 이미지 추가

mozzi34 committed:
feat: 랜딩페이지 및 전체 섹션 추가

mozzi34 committed:
refactor: Session3 탭 콘텐츠 컴포넌트 추가 (스터디 보드, 실시간 수업 등)

mozzi34 committed:
style: Tailwind size 클래스 통일 및 Session1 버튼 스타일 정리

- px 기반 spacing 및 text size → Tailwind 기본 scale로 통일
- Session1 버튼을 cva 유틸로 구조화

mozzi34 committed:
style: 랜딩페이지 히어로 이미지 교체

mozzi34 committed:
feat: 라우트 상수 정의

mozzi34 committed:
chore: src/constants, src/features의 .gitkeep 파일 삭제

mozzi34 committed:
refactor: ProblemCard로 타입 및 변수명 변경

osh6006 commented on src/features/home/index.tsx:
혹시 그냥 메인 페이지 안에 이 컴포넌트 4개를 넣어도 될 것 같은데,

이 HomePage 라는 컴포넌트를 만든 이유를 알 수 있을까요??

mozzi34 replied:
제가 생각했을 때 page는 최대한 가볍게 유지하고, 내부 구현은 하위에서 관리하는 게 유지보수나 확장성 측면에서 더 유리하다고 생각해서 구조를 나눴습니다.
다만 지금처럼 단순한 랜딩 페이지에서는 이렇게까지 분리하는 게 조금 과했을 수도 있겠다는 생각이 들어 구조를 수정했습니다!

osh6006 replied:
네네 이미 HomePage라는 컴포넌트가 `app/page.tsx`라는 컴포넌트이기 때문에 이렇게 하는게 맞는 것 같습니다!

mozzi34 replied:
넵~! 감사합니다~!

nononcrust commented on src/features/home/components/Session1.tsx:
```diff
@@ -0,0 +1,79 @@
+    <section className="mt-[136px] mb-[220px] max-w-[1385px]">
+      <div className="flex-co flex gap-10 md:flex-row">
+        <div className="mx-auto max-w-5xl">
+          <p className="text-[2.5rem] leading-[1.6em] font-normal tracking-[-0.05em]">
```
문서구조상 h1이 좋을것같아요

nononcrust commented on src/features/home/components/Session1.tsx:
```diff
@@ -0,0 +1,79 @@
+          </div>
+
+          <div className="mt-[282px] flex gap-3">
+            <Link href={ROUTE.LOGIN}>
```
a태그랑 button 태그가 중첩되어있는것보다 a태그만 사용하는게 좋을 것 같아요

nononcrust commented on src/features/home/components/Session1.tsx:
```diff
@@ -0,0 +1,79 @@
+
+export default function Session1() {
+  return (
+    <section className="mt-[136px] mb-[220px] max-w-[1385px]">
```
max-w-[1385px]가 컴포넌트 마다 중복으로 나타나고있어서 모듈화하면 좋을것같아요

mozzi34 replied:
max-w-[1385px]가 여러 컴포넌트에서 반복되어 모듈화할 수 있겠다는 피드백을 받고 관련해서 고민해 봤는데요,
현재는 Session1, Session2가 같은 max-width 값을, Session3, Session4는 또 다른 값을 갖고 있어서 각각 두 번씩 반복되긴 하지만, 이 정도의 반복을 기준으로 컴포넌트를 분리했을 때 얻는 이점이 클지는 잘 모르겠습니다....
구조적으로 더 확장되거나 의미가 뚜렷해진다면 나누는 게 맞을 것 같은데, 지금 상태에서는 오히려 간단하게 유지하는 게 더 낫지 않을까요?

mozzi34 replied:
엇 혹시 globals.css나 Tailwind를 사용한 모듈화를 말씀하신 걸까요?

nononcrust replied:
아뇨 랜딩페이지 관련 스타일이라 글로벌로 뺴는건 아닌것같고 말씀주신대로 1385랑 1344가 섞여있어서 분리하기 애매한 부분은 있는것 같네요!

nononcrust commented on src/features/home/components/Session3.tsx:
```diff
@@ -0,0 +1,103 @@
+        {tabs.map((tab, index) => (
+          <button
+            key={index}
+            onClick={() => setActiveTab(index as TabIndex)}
```
tabs마다 value를 정의하면 index를 리터럴로 단언하는 부분을 없앨 수 있을 것 같아요

nononcrust commented on src/features/home/components/Session4.tsx:
```diff
@@ -0,0 +1,103 @@
+            key={index}
+            className="cursor-pointer"
+          >
+            <div
```
클릭 가능한 요소는 button 태그를 사용하는게 좋을것같아요

nononcrust commented on src/features/home/components/Session4.tsx:
```diff
@@ -0,0 +1,103 @@
+                <p className="text-base leading-[27px] font-normal tracking-[-0.05em]">
+                  {Array.isArray(item.answer)
+                    ? item.answer.map((line, i) => (
+                        <span key={i}>
```
span과 br로 각 줄을 분리하는것보단 p태그를 사용하는게 좋을것같아요

nononcrust commented on src/features/home/components/Session3.tsx:
```diff
@@ -0,0 +1,103 @@
+      <div className="mt-[77px] mr-[36px] flex justify-center">
+        <TabSection />
+      </div>
+      <div className="mx-auto mt-[117px] h-[1px] w-full max-w-[1344px] bg-gray-200" />
```
구분선이 오른쪽으로 삐져나와있는것같아요

nononcrust commented on src/features/home/components/Session3.tsx:
```diff
@@ -0,0 +1,103 @@
+export default function Session3() {
+  return (
+    <section className="mt-[179px] w-full">
+      <div className="mx-auto flex max-w-[1385px] flex-col-reverse items-center md:flex-row">
```
이부분은 피그마상 1344px인것 같아요 md:는 사용하신 이유가 뭘까요?

nononcrust commented on src/features/home/components/Session1.tsx:
```diff
@@ -0,0 +1,79 @@
+  },
+});
+
+export default function Session1() {
```
컨벤션에 따라 명시적 export로 내보내는게 좋을것 같아요

mozzi34 replied:
모든 피드백 반영했습니다 😊
구조 분리, 명시적 export, 레이아웃 수정 등 모두 적용 완료했습니다.
꼼꼼한 리뷰 감사합니다!

mozzi34 committed:
refactor: 랜딩 페이지에서 HomePage 제거 및 구조 변경

mozzi34 committed:
refactor: 시맨틱 마크업 및 코드 컨벤션 개선

- 문서 구조에 맞게 h1 태그로 수정
- a 태그 내부 중첩된 button 제거
- span/br → p 변경
- 컴포넌트 export 방식을 명시적 export로 변경

mozzi34 committed:
refactor: tabs 배열에 value 추가, CSS 정리

mozzi34 committed:
rename: 컴포넌트 파일명 케밥 케이스로 통일 (Session1 → session-1)

mozzi34 committed:
fix: 누락된 page.tsx 파일 추가

mozzi34 committed:
chore: 이미지 경로를 public/home 하위로 이동, 참조 경로 수정
